yjit/src/codegen.rs

```diff
@@ -68,10 +68,25 @@ impl JITState {
         }
     }
 
+    pub fn new_with_opcode(opcode: usize) -> Self {
```

Comment:
So the opcode field and the PC in the JIT state have been used to keep track of where we are during compilation. I don't think it should be set in a constructor. IMO you should create a JITState object and then set the opcode, because using this for testing purposes is very much a special case.